src/pykeen/nn/representation.py

```diff
@@ -1570,15 +1570,28 @@ def __init__(
             If the backfill representation is initialized within this constructor,
             it will receive the base representation's shape.
         """
+        base_ids = sorted(set(base_ids))
+
+        # If the maximum id is not greater than the maximum base id, there will be an error. So, we check and raise explicitly here.
+        # If the max ID is less than the length of the base IDs, the representation_resolver.make() will fail since a tensor with negative dimensions cannot be created.
+        # If the max ID is not greater than the maximum base ID, IndexError will result on `assignment[base_ids, 0] = 0`
+        # This is because the `assignment` tensor created from max_id will not have enough space to accomodate all the base_ids.
+        if max_id <= max(base_ids):
+            raise ValueError(f"{max_id=}: Value must be greater than the largest base_id ({max(base_ids)=})")
+
         # import here to avoid cyclic import
         from . import representation_resolver
 
-        base_ids = sorted(set(base_ids))
         base = representation_resolver.make(base, base_kwargs, max_id=len(base_ids))
+
+        backfill_max_id = max_id - base.max_id
+        if backfill_max_id == 0:
+            raise BackfillZeroMaxIDError
```

Comment:
This is not necessarily an error: The backfill representation will just be a equivalent to the base representation.

But maybe it is worth to emit a warning instead.

Reply:
that's what I thought - it would be pretty inefficient to do all of the intermediate backfill representation operations if you could have directly used the original representation

Reply:
b2abee2b